storage/core/src/main/java/com/hortonworks/registries/storage/impl/jdbc/config/HikariConfigFactory.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright 2016 Hortonworks.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package com.hortonworks.registries.storage.impl.jdbc.config;
+
+import com.hortonworks.registries.storage.common.DatabaseType;
+import com.hortonworks.registries.storage.common.util.Constants;
+import com.zaxxer.hikari.HikariConfig;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+public class HikariConfigFactory {
```

Comment:
Will you remove the `HikariBasicConfig` class and replace it's usages with this class?

Reply:
HikariBasicConfig is predominately used by test classes. May be it makes more sense to move it to test package, I will refactor this in a separate PR.